app/eventyay/api/views/order.py

```diff
@@ -1247,8 +1247,14 @@ def confirm(self, request, **kwargs):
         return self.retrieve(request, [], **kwargs)
 
     @action(detail=True, methods=['POST'])
+    @transaction.atomic
```

Comment:
Putting `transaction.atomic` here doesn't work, because this `refund` function already handle an `PaymentException`:

```py
try:
    r.payment_provider.execute_refund(r)
except PaymentException as e:
```

Read the [Avoid catching exceptions inside atomic!](https://docs.djangoproject.com/en/6.0/topics/db/transactions/) note.